File: DietitianHub-Backend/src/main/java/gr/unipi/thesis/dimstyl/dtos/api/AppointmentsResponse.java
package gr.unipi.thesis.dimstyl.dtos.api;

import lombok.Builder;

import java.util.List;

@Builder
public record AppointmentsResponse(List<ApiAppointmentDto> scheduledAppointments,
                                   List<ApiAppointmentDto> pendingAppointments,
                                   List<ApiAppointmentDto> completedAppointments,
                                   List<ApiAppointmentDto> declinedAppointments,
                                   List<ApiAppointmentDto> cancelledAppointments) {
}

File: DietitianHub-Android/app/src/main/java/gr/unipi/thesis/dimstyl/data/models/AppointmentsResponse.kt
package gr.unipi.thesis.dimstyl.data.models

data class AppointmentsResponse(
    val scheduledAppointments: List<Appointment>,
    val pendingAppointments: List<Appointment>,
    val completedAppointments: List<Appointment>,
    val declinedAppointments: List<Appointment>,
    val cancelledAppointments: List<Appointment>
)
